crates/common/src/lib.rs

```diff
@@ -135,11 +139,32 @@ macros::starkhash251::deserialization!(StorageAddress);
 #[derive(Copy, Clone, PartialEq, Eq, Deserialize, Serialize, PartialOrd, Ord)]
 pub struct StorageValue(pub Felt);
 
-/// A commitment root of the global StarkNet state. This is the entry-point
-/// for the global state at a specific point in time via the global state tree.
+/// A global state commitment root for Starknet state starting from Starknet v0.11.0,
+/// when Declare v2 transactions became part of the state commitment.
 #[derive(Copy, Clone, PartialEq, Eq, Deserialize, Serialize)]
 pub struct StateCommitment(pub Felt);
 
+impl From<(StorageCommitment, ClassCommitment)> for StateCommitment {
+    fn from((storage_commitment, class_commitment): (StorageCommitment, ClassCommitment)) -> Self {
+        if class_commitment == ClassCommitment::ZERO {
+            Self(storage_commitment.0)
+        } else {
+            // FIXME 0.11.0 make sure which hash to use Pedersen or Poseidon
+            StateCommitment(stark_hash::stark_hash(
+                storage_commitment.0,
+                class_commitment.0,
+            ))
+        }
+    }
+}
```

Comment:
I'm not sure how I feel hiding the code inside a `From` implementation (which I feel is more a transformation of the data rather than something as expensive as a hash function).

Maybe `StateCommitment::calculate(StorageCommitment, ClassCommitment)` instead?